src/gui/share.h

```diff
@@ -0,0 +1,205 @@
+#ifndef SHARE_H
+#define SHARE_H
+
+#include "accountfwd.h"
+
+#include <QObject>
+#include <QDate>
+#include <QString>
+#include <QList>
+#include <QSharedPointer>
+#include <QUrl>
+
+namespace OCC {
+
+class Share : public QObject {
+    Q_OBJECT
+
+public:
+
+    /*
+     * Constructor for link shares
+     */
+    explicit Share(AccountPtr account,
+                   const QString& id,
+                   const QString& path,
+                   int shareType,
+                   int permissions,
```

Comment:
It'd be nice if these had the ShareType and Permission types. Consider using Q_DECLARE_FLAGS on the enums (at least Permissions).